image-loader/app/lib/imaging/MimeTypeDetection.scala

```
@@ -2,12 +2,18 @@ package lib.imaging
 
 import java.io.File
 
+import com.gu.mediaservice.model.{MimeType, UnsupportedMimeTypeException}
 import org.apache.tika.Tika
+import com.gu.mediaservice.lib.logging._
+
+import scala.util.{Failure, Success, Try}
 
 object MimeTypeDetection {
   val tika = new Tika()
 
-  def guessMimeType(file: File): Option[String] =
-    Option(tika.detect(file))
-
+  def guessMimeType(file: File): Either[UnsupportedMimeTypeException, MimeType] = Try(MimeType(tika.detect(file))) match {
```

This allows us to fail fast on the `Left`.